libs/net/dal/Migrations/1.0.87/Up/PostUp/00-ReportTemplate.sql

```diff
@@ -0,0 +1,218 @@
+        else
+        {
+          var loopCount = 0;
+          var allContent = Content.ToArray();
+          foreach (var section in Sections)
+          {
+            var sectionContent = section.Value.Content.ToArray();
+            if (section.Value.IsEnabled &&
+              (sectionContent.Length > 0 ||
+              !section.Value.Settings.HideEmpty ||
+              (section.Value.Settings.SectionType == ReportSectionType.TableOfContents)
+                && Content.Any()))
+            {
+              <hr style="width: 100%; display: inline-block;" />
+              <div style="@pageBreak">
+                @if (!String.IsNullOrEmpty(section.Value.Settings.Label))
+                {
+                  <h2 id="section-@section.Value.Id" styled="margin: 0; padding: 0;">@section.Value.Settings.Label<a name="section-@section.Value.Id"></a></h2>
+                }
+                @if (!String.IsNullOrEmpty(section.Value.Description))
+                {
+                  <p>@section.Value.Description</p>
+                }
+                @* Charts *@
+                @if (section.Value.Settings.ChartsOnTop && section.Value.Settings.ShowCharts)
+                {
+                  foreach (var chart in section.Value.ChartTemplates)
+                  {
+                    <img alt="@chart.SectionSettings.AltText" src="@chart.Uid" />
+                  }
+                }
+                @if (section.Value.Settings.SectionType == ReportSectionType.Content)
+                {
+                  @* Section Table of Contents *@
+                  var indexPosition = 0;
+                  if (section.Value.Settings.ShowHeadlines)
+                  {
+                      @if (section.Value.Settings.GroupBy != "")
+                      {
+                        var contentGroupings = sectionContent.GetContentGroupings(section.Value.Settings.GroupBy);
+                        <ul style="margin:0; margin-left: 25px; padding:0;">
+                        @foreach(KeyValuePair<string, List<long>> contentGroup in contentGroupings)
+                        {
+                        <li>@section.Value.Settings.GroupBy = @contentGroup.Key</li>
+                          <ul style="margin:0; margin-left: 25px; padding:0;">
+                            @foreach(long contentId in contentGroup.Value)
+                            {
+                              var content = sectionContent.FirstOrDefault(c => c.Id == contentId);
+                              var headline = $"{(Settings.Headline.ShowSentiment ? content.GetSentimentIcon("{0} - ") : "")}{content.Headline}{(Settings.Headline.ShowShortName && !String.IsNullOrEmpty(content.Source?.ShortName) ? $" - {content.Source?.ShortName}" : (Settings.Headline.ShowSource ? $" - {content.OtherSource}": ""))}{(Settings.Headline.ShowPublishedOn ? $" - {content.PublishedOn?.AddHours(utcOffset):dd-MMM-yyyy}" : "")}";
+                              var headLineLink = loopCount + indexPosition;
```

Comment:
This doesn't feel right.  If `loopCount` increments per section and `indexPosition` is the index of the article in the section, wont the `headLineLink` end up being the same for 3rd article in the 1st section (`3 + 1 = 4`) and the 2nd article in the second section (`2 + 2 = 4`) ?